cloudamqp/resource_cloudamqp_integration_metric_prometheus.go

```diff
@@ -0,0 +1,267 @@
+							Required:    true,
+							Sensitive:   true,
+							Description: "Azure Application Insights Connection String",
+						},
+						"tags": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "tags. E.g. env=prod,region=europe",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceIntegrationMetricPrometheusCreate(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	var (
+		api     = meta.(*api.API)
+		intName string
+		params  = make(map[string]any)
+	)
+
+	// Check which integration type is configured
+	if newrelicList := d.Get("newrelic_v3").(*schema.Set).List(); len(newrelicList) > 0 {
+		intName = "newrelic_v3"
+		newrelicConfig := newrelicList[0].(map[string]any)
+		params["api_key"] = newrelicConfig["api_key"]
+		if tags := newrelicConfig["tags"]; tags != nil && tags != "" {
+			params["tags"] = tags
+		}
+	} else if datadogList := d.Get("datadog_v3").(*schema.Set).List(); len(datadogList) > 0 {
+		intName = "datadog_v3"
+		datadogConfig := datadogList[0].(map[string]any)
+		params["api_key"] = datadogConfig["api_key"]
+		if region := datadogConfig["region"]; region != nil && region != "" {
+			params["region"] = region
+		}
+		if tags := datadogConfig["tags"]; tags != nil && tags != "" {
+			params["tags"] = tags
+		}
+	} else if azureMonitorList := d.Get("azure_monitor").(*schema.Set).List(); len(azureMonitorList) > 0 {
+		intName = "azure_monitor"
+		azureMonitorConfig := azureMonitorList[0].(map[string]any)
+		params["connection_string"] = azureMonitorConfig["connection_string"]
+		if tags := azureMonitorConfig["tags"]; tags != nil && tags != "" {
+			params["tags"] = tags
+		}
+	}
+
+	if intName == "" {
+		return diag.Errorf("no integration configuration provided")
+	}
+
+	data, err := api.CreateIntegration(ctx, d.Get("instance_id").(int), "metrics", intName, params)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	if data["id"] != nil {
+		d.SetId(data["id"].(string))
+	}
+
+	return resourceIntegrationMetricPrometheusRead(ctx, d, meta)
+}
+
+func resourceIntegrationMetricPrometheusRead(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	if strings.Contains(d.Id(), ",") {
+		tflog.Info(ctx, fmt.Sprintf("import resource with identifier: %s", d.Id()))
+		s := strings.Split(d.Id(), ",")
+		d.SetId(s[0])
+		instanceID, _ := strconv.Atoi(s[1])
+		d.Set("instance_id", instanceID)
+	}
+	if d.Get("instance_id").(int) == 0 {
+		return diag.Errorf("missing instance identifier: {resource_id},{instance_id}")
+	}
+
+	api := meta.(*api.API)
+	data, err := api.ReadIntegration(ctx, d.Get("instance_id").(int), "metrics", d.Id())
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	// Handle resource drift and trigger re-creation if resource been deleted outside the provider
+	if data == nil {
+		d.SetId("")
+		return nil
+	}
+
+	// Clear all blocks first
+	d.Set("newrelic_v3", nil)
+	d.Set("datadog_v3", nil)
```

Comment:
Also clear `azure_monitor` block?

```suggestion
	// Clear all blocks first
	d.Set("newrelic_v3", nil)
	d.Set("datadog_v3", nil)
	d.Set("azure_monitor", nil)
```